src/robot/libraries/BuiltIn.py

```diff
@@ -2220,8 +2224,17 @@ def wait_until_keyword_succeeds(self, retry, retry_interval, name, *args):
             if count <= 0:
                 raise ValueError('Retry count %d is not positive.' % count)
             message = '%d time%s' % (count, s(count))
-        retry_interval = timestr_to_secs(retry_interval)
+        match = re.search(r'(strict *: *)', retry_interval)
+        if match:
+            strict_interval = True
+            retry_interval = retry_interval.lstrip(match.group())
+        else:
+            strict_interval = False
+
+        retry_interval = float(timestr_to_secs(retry_interval))
```

Comment:
Some problems here:

1. `retry_interval` can be given also as an integer or float in which case `re.search` will fail. We clearly don't have any tests for such behavior but I'm absolutely sure someone has used that functionality.
2. Using regexps is an overkill when we just to check does the string start with something and `re.search` is actually wrong because it can find the match anywhere, not only from the beginning.
3. Would be good to support `strict` case-insensitively like we support most similar markers.
4. Unnecessary empty line. We don't generally have them inside methods and for consistency reasons shouldn't use here either.
5. `time_str_to_secs` returns a floating point number so `float()` isn't needed.

This (untested) code ought to handle all above problems:
```python
if is_string(retry_interval) and retry_interval.lower().startswith('strict:'):
    retry_interval = retry_interval.split(':', 1).strip()
    strict_interval = True
else:
    strict_interval = False
retry_interval = timestr_to_secs(retry_interval)
```